python/lsst/ts/externalscripts/maintel/tma/random_walk_and_take_image_gencam.py

```diff
@@ -0,0 +1,583 @@
+                )
+
+            # Slew to the next target
+            await self.slew_and_track(
+                data.az, data.el, target_name=f"random_walk_{data.counter}"
+            )
+
+            # Take data in sync with all the GenericCameras
+            await self.take_data()
+
+    def set_metadata(self, metadata):
+        """Set estimated duration of the script."""
+        metadata.duration = self.config.total_time
+
+    async def slew_and_track(self, az, el, target_name=None):
+        """Slew to and track a new target.
+
+        Parameters
+        ----------
+        az : `float`
+            Azimuth in hour angle.
+        el :  `float`
+            Elevation in degrees.
+        target_name : `str`, optional
+            Target name. Default: None
+        """
+        target_name = target_name if target_name else "random_walk"
+        radec = self.tcs.radec_from_azel(az, el)
+
+        ra = float(radec.ra.hour)
+        dec = float(radec.dec.deg)
+
+        self.log.info(
+            f"Slew and track target_name={target_name}; "
+            f"ra={ra}, dec={dec}; "
+            f"rot={self.config.rot_value}; rot_type={self.config.rot_type}; "
+        )
+
+        await self.tcs.slew_icrs(
+            ra=ra,
+            dec=dec,
+            rot=self.config.rot_value,
+            rot_type=self.config.rot_type,
+            target_name=target_name,
+        )
+
+        if self.config.track_for > 0.0:
+            self.log.info(f"Tracking for {self.config.track_for}s .")
+            await self.tcs.check_tracking(self.config.track_for)
+            if self.config.stop_when_done:
+                self.log.info("Tracking completed. Stop tracking.")
+                await self.tcs.stop_tracking()
+            else:
+                self.log.info("Tracking completed.")
+
+    async def stop_tracking(self):
+        """Execute stop tracking on MTCS."""
+        await self.tcs.stop_tracking()
+
+    async def take_data(self):
+        """Take data while making sure MTCS is tracking."""
+
+        tasks = [
+            asyncio.create_task(self._take_data()),
+            asyncio.create_task(self.tcs.check_tracking()),
+        ]
+
+        await self.tcs.process_as_completed(tasks)
+
+    @property
+    def tcs(self):
+        return self._mtcs
+
+    async def track_target_and_setup_instrument(self):
+        """Track target and setup instrument in parallel."""
+
+        self.tracking_started = True
+
+        await self.tcs.slew_icrs(
+            ra=self.config.ra,
+            dec=self.config.dec,
+            rot=self.config.rot_sky,
+            rot_type=RotType.Sky,
+            target_name=self.config.name,
+            az_wrap_strategy=self.config.az_wrap_strategy,
+        )
+
+    async def wait_for_dome_in_position(self):
+        """Wait until the dome is in position"""
+        await asyncio.sleep(20)
+        try:
+            azMotion = await self.tcs.rem.mtdome.evt_azMotion.aget()
```

Comment:
I think this logic has a few issues:
* Race conditions: aget may return stale information -- the dome may not even start to move before you call this method.
* Its unwise to call get repeatedly, yet this method is called repeatedly.

To avoid the race condition and make sure that the dome state is fresh, I strongly suggest the following:
* Delete this method. Move the logic into move_dome.
* In move_dome flush evt_azMotion before commanding motion.
* Ditch the call to aget; just loop over calls to evt_azMotion.next(flush=False) as you do below.

The point of combining this method into move_dome is that users cannot accidentally call it without commanding a dome move (which would make it hang forever if you only call next(flush=False)).